openapi.yaml

```diff
@@ -16,175 +16,301 @@ host: api.oregonstate.edu
+  /notes/{noteID}:
+    parameters:
+      - name: noteID
+        in: path
+        type: string
+        required: true
+    patch:
+      summary: Edit an existing note
+      tags:
+        - notes
+      operationId: patchNoteByID
+      parameters:
+        - in: body
+          name: requestBody
+          schema:
+            type: object
+            properties:
+              note:
+                $ref: '#/definitions/note'
+              permissions:
+                $ref: '#/definitions/permissions'
+              dateCreated:
+                $ref: '#/definitions/dateCreated'
+      responses:
+        200:
+          description: Successfully updated the note
+          schema:
+            $ref: '#/definitions/NoteResult'
+        400:
+          description: Invalid patch body
+          schema:
+            $ref: '#/definitions/Errors'
+        404:
+          description: Note not found
+          schema:
+            $ref: '#/definitions/Errors'
+        500:
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+    delete:
+      summary: Delete an existing note
+      tags:
+        - notes
+      operationId: deleteNoteByID
+      responses:
+        204:
+          description: Successfully deleted the note
+        404:
+          description: Note not found
+          schema:
+            $ref: '#/definitions/Errors'
+  /notes/{noteID}/versions:
+    parameters:
+      - name: noteID
+        in: path
+        type: string
+        required: true
+    get:
+      summary: Get version history of a note
+      tags:
+        - notes
+      operationId: getVersions
       responses:
         200:
           description: Successful response
           schema:
-            $ref: '#/definitions/PetResult'
+            $ref: '#/definitions/VersionHistoryResult'
         404:
-          description: PetResource not found
+          description: Note not found
           schema:
             $ref: '#/definitions/Errors'
         500:
-          description: Internal Server Error
+          description: Internal server error
           schema:
             $ref: '#/definitions/Errors'
-parameters:
-  authorization:
-    name: Authorization
-    in: header
-    type: string
-    required: true
-    description: '"Bearer [token]" where [token] is your OAuth2 access token'
-  pageNumber:
-    name: page[number]
-    in: query
-    type: integer
-    required: false
-    description: Page number of results
-    default: 1
-    minimum: 1
-  pageSize:
-    name: page[size]
-    in: query
-    type: integer
-    required: false
-    description: Number of results to return
-    default: 25
-    maximum: 500
-    minimum: 1
 definitions:
-  Meta:
+  noteID:
+    type: string
+    description: A unique ID of a note concatenating the student ID that the note is for with a random string
+    example: 123456789-abc123
+  note:
+    type: string
+    description: The contents of the note
```

Comment:
Do we want to limit the length of `note` eventually? If so, we should return a `400` if the length of `note` in `POST` or `PATCH` request body is too long and we need to add some description here.

Reply:
This sounds good but we should probably discuss if we want to limit the length of `note` and if so, what the limit should be.

Reply:
We discussed this in our team meeting, so let's adding some constraint to this field.